### PR TITLE
Create temp directories if they do not exist

### PR DESCRIPTION
**Pull request by geedo0**
*Issue #, if available:*
ACCP-72

*Description of changes:*
This applies some changes necessary to improve unit test coverage, fixes a bug
around symlink handling, and improves error visibility.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by WillChilds-Klein** [comment]
> fixes a bug around symlink handling

what was the bug?

**Review by WillChilds-Klein — APPROVED** [review]

**Comment by geedo0** [comment]
> what was the bug?

If the tmpdir is a symlink, ACCP fails to load the library.

File: src/com/amazon/corretto/crypto/provider/Loader.java [hunk 1]
```diff
@@ -269,17 +270,22 @@ private synchronized static Path createTmpFile(final String prefix, final String
                 final Path tmpFile = tmpDir.resolve(fileName.toString());
 
                 try {
+                    if (!Files.exists(tmpDir)) {
+                        // Creating tmpDir in case it's been deleted
+                        Files.createDirectories(tmpDir, permissions);
```
[the remark this answers is not in the document]
Comment (geedo0) [reply src/com/amazon/corretto/crypto/provider/Loader.java:275]: Yes, but it's a lot of hassle. You'd have to figure out all the setup/teardown for the File I/O, Java properties, POSIX permissions, symlinks... None of this has been developed for this class yet.

[the remark this answers is not in the document]
Comment (WillChilds-Klein) [reply src/com/amazon/corretto/crypto/provider/Loader.java:275]: probably not worth investing that into v1 given it's on the deprecation path.

**Review by WillChilds-Klein — APPROVED** [review]

**Pull request merged**